unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -456,6 +501,66 @@ findShallowInBranch codebase b = do
       ++ branchEntries
       ++ patchEntries
 
+-- | Find all definitions and children reachable from the given 'V2Branch.Branch',
+-- Note: this differs from 'lsBranch' in that it takes a shallow v2 branch,
+-- As a result, it omits definition counts from child-namespaces in its results,
+-- but doesn't require loading the entire sub-tree to do so.
+lsShallowBranch ::
```

Comment:
This is effectively a copy of `lsBranch` but uses 'lazy-loaded' branches.

The deep version is used by the `ls` command because it's nice for ls to show the number of definitions inside a namespace, but it isn't efficient on shallow branches, so this new version omits definition counts but provides the necessary results for the `browse` API.

```
.> ls
  1. base/        (23013 definitions)
```